src/goReferences.ts

```diff
@@ -19,7 +19,7 @@ export class GoReferenceProvider implements vscode.ReferenceProvider {
 		});
 	}
 
-	private doFindReferences(document: vscode.TextDocument, position: vscode.Position, options: { includeDeclaration: boolean }, token: vscode.CancellationToken): Thenable<vscode.Location[]> {
+	public doFindReferences(document: vscode.TextDocument, position: vscode.Position, options: { includeDeclaration: boolean }, token: vscode.CancellationToken): Thenable<vscode.Location[]> {
```

Comment:
I am guessing the reason you are using `doFindReferences` instead of `provideReferences` is because the latter does a `saveAll` before finding the references.

My another guess for why `saveAll` was put in place was because `guru` was made to run on saved files.

Now that we have `guru` working on unsaved files as well (with the -modified flag), lets remove the `saveAll` that happens in `provideReferences` and you can use it here.